src/bot.ts

```diff
@@ -499,6 +503,83 @@ export class DiscordBot {
         return rooms.map((room) => room.matrix.getId());
     }
 
+    public async HandleMatrixKickBan(
+        roomId: string, kickeeUserId: string, kicker: string, kickban: "leave"|"ban",
+        previousState: string, reason?: string,
+    ) {
+        const restore = kickban === "leave" && previousState === "ban";
+        const client = await this.clientFactory.getClient(kicker);
+        let channel: Discord.Channel;
+        try {
+            channel = await this.GetChannelFromRoomId(roomId, client);
+        } catch (ex) {
+            log.error("Failed to get channel for ", roomId, ex);
+            return;
+        }
+        if (channel.type !== "text") {
+            log.warn("Channel was not a text channel");
+            return;
+        }
+        const tchan = (channel as Discord.TextChannel);
+        const kickeeUser = (await this.GetDiscordUserOrMember(
+            new MatrixUser(kickeeUserId.replace("@", "")).localpart.substring("_discord".length),
```

Comment:
The MatrixUser should do all the validation as well as fetching us a localpart. I'd rather use the bridge libs method of getting a localpart rather than doing our own.

Reply:
fair enough